### DISCUSSION TITLE
TileLayer MapView Repeat (false) Whitespace?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5561] [Q&A]
Would there be an approach which would allow for the [MapView](https://deck.gl/docs/api-reference/core/map-view) `repeat` property when set to `false` to eliminate the whitespace seen when using a [TileLayer](https://deck.gl/docs/api-reference/geo-layers/tile-layer)?

![image](https://user-images.githubusercontent.com/1843941/110218713-2d444400-7e89-11eb-963c-e2d5f299450c.png)

I have a use-case where users do not want to see the map duplicated _(i.e., _repeat: true_)_.

Thanks in advance for any ideas!

**Comment by ibgreen** [comment]
I probably do not understand what you are trying to achieve, but you would like to eliminate white space when zoomed out.
-  You could set maxZoom props on controller or cap zoom on view to prevent zooming out.
- You could create two Views on top of each other, one with `repeat: true` for a base layer that covers the screen, and another View with `repeat: false` for your visualizations.

**Comment by Pessimistress** [comment]
You can [add constraints to the view state](https://deck.gl/docs/developer-guide/interactivity#add-constraints-to-view-state). You will need to use [WebMercatorViewport](https://deck.gl/docs/api-reference/core/web-mercator-viewport) to find the range of longitude at a given zoom level.

**Reply by jfuehner** [reply]
@Pessimistress 

I tried using the technique you provided me with long ago in [#4234](https://github.com/uber/deck.gl/issues/4234)to limit the bounds but that didn't seem to yield the desired results.

I am unclear how the zoom level would factor in? Any examples I could look at?

**Comment by TajBanana** [comment]
Hi all, i am currently facing a similar problem mentioned in this issue.
What I am trying to achieve is to limit the max range of WebMercatorViewPort to 360° in longitude while enabling repeat world, so that the user can zoom in and out of the map till the max extent in longitude range (360°).

If i were to follow the example #4234, it will restrict panning and not be continuous.
i was wondering if anyone has met with this issue and can help me with it.

![image](https://github.com/visgl/deck.gl/assets/95390298/5259936f-d1d3-4ff4-bc66-bc5ace1505aa)
![image](https://github.com/visgl/deck.gl/assets/95390298/872b9875-5928-4993-9928-afbf55d6a922)
